catroid/src/main/java/org/catrobat/catroid/content/Project.java

```diff
@@ -133,41 +131,14 @@ public Project(SupportProject supportProject, Context context) {
 		projectLists = supportProject.dataContainer.projectLists;
 
 		DataContainer container = new DataContainer(this);
-		removeInvalidVariablesAndLists(supportProject.dataContainer);
-		container.setSpriteVariablesForSupportContainer(supportProject.dataContainer);
+		container.setSpriteUserData(supportProject.dataContainer);
 
 		Scene scene = new Scene(context.getString(R.string.default_scene_name, 1), this);
 		scene.setDataContainer(container);
 		scene.getSpriteList().addAll(supportProject.spriteList);
 		sceneList.add(scene);
 	}
 
-	public void removeInvalidVariablesAndLists(BaseDataContainer dataContainer) {
-		if (dataContainer == null) {
-			return;
-		}
-
-		if (dataContainer.spriteListOfLists != null) {
-			Iterator listIterator = dataContainer.spriteListOfLists.keySet().iterator();
-			while (listIterator.hasNext()) {
-				Sprite sprite = (Sprite) listIterator.next();
-				if (sprite == null) {
-					listIterator.remove();
-				}
-			}
-		}
-
-		if (dataContainer.spriteVariables != null) {
-			Iterator variablesIterator = dataContainer.spriteVariables.keySet().iterator();
-			while (variablesIterator.hasNext()) {
-				Sprite sprite = (Sprite) variablesIterator.next();
-				if (sprite == null) {
-					variablesIterator.remove();
-				}
-			}
-		}
-	}
-
```

Comment:
There are ~70 programs having `null` as key. This code has been added in a hotfix https://github.com/Catrobat/Catroid/commit/626e5fdb4ca60bf6f7835301defcbab475a64c53, maybe that's why the number is so low.

Should we keep it? Maybe use [Map.remove](https://developer.android.com/reference/java/util/Map.html#remove(java.lang.Object))?
This code is a good indicator that the DataContainer should prevent from storing null Sprites (future improvements :wink:)

Reply:
sure, I can do that. btw: https://github.com/Catrobat/Catroid/pull/2864/files#diff-cb1b43b9b2d00fb9b796dba270e0093e prevents you from adding UserData to a null key.

Reply:
Awesome! :party:

Reply:
I have moved the null entry removal just below that [that] (I wasn't able to get the exact line - I'm sure I could have but it took me too long).